test/integ/test_cursor.py

```diff
@@ -975,6 +975,15 @@ def test_fetch_out_of_range_timestamp_value(conn, result_format):
             cur.fetchone()
 
 
+@pytest.mark.skipolddriver
+def test_null_in_non_null(conn):
+    with conn() as cnx:
+        cur = cnx.cursor()
+        cur.execute("create table null_in_non_null(bar char not null)")
+        with pytest.raises(errors.IntegrityError):
```

Comment:
Please add a match regex here, that way we also verify that the error message hasn't been screwed up in the future.